core/src/main/scala/com/yahoo/maha/core/dimension/Dimension.scala

```diff
@@ -969,6 +969,8 @@ trait PublicDimension extends PublicTable {
 
   def dimLevel: DimLevel
 
+  var secondaryDimLevel: Int = 0
```

Comment:
cant put secondaryDimLevel  in the immutable registry, instead use the dim bundle or dim candidate to store the query level vars.